File: Week 6/pset6/cash.py

# cash in python

from cs50 import get_float


# getting change
while True:
    x = get_float("Change owed: ")
    if x > 0:
        break

# rouding it
x = round(x * 100)

# initializing number of coins
coins = 0

# getting number of coins
while x > 0:
    if x >= 25:
        x -= 25
        coins += 1
    elif x >= 10:
        x -= 10
        coins += 1
    elif x >= 5:
        x -= 5
        coins += 1
    elif x >= 1:
        x -= 1
        coins += 1

# printing result
print(coins)

File: Week 1/pset1/cash/cash.c

#include <stdio.h>
#include <cs50.h>
#include <math.h>


int main(void)

{
    float i;                                        // declaring variables
    int change;
    int quarters;
    int remainder_quarters;
    int dimes;
    int remainder_dimes;
    int nickels;
    int remainder_nickels;
    int pennies;
    int coins;

    do
    {
        i = get_float("Change owed: ");             // getting change
    }
    while (i <= 0);


    change = round(i * 100);                        // rounding to cents

    quarters = change / 25;                         // finding number of quarters
    remainder_quarters = change % 25;
    dimes = remainder_quarters / 10;                // finding number of dimes
    remainder_dimes = remainder_quarters % 10;
    nickels = remainder_dimes / 5;                  // finding number of nickels
    remainder_nickels = remainder_dimes % 5;
    pennies = remainder_nickels / 1;                // finding number of pennies
    
    coins = quarters + dimes + nickels + pennies;
    
    printf("%i\n", coins);

}
